htdocs/societe/class/api_thirdparties.class.php

```diff
@@ -467,6 +467,68 @@ public function setThirdpartyPriceLevel($id, $priceLevel)
 		return $this->_cleanObjectDatas($this->company);
 	}
 
+	/**
+	 * Add a customer representative to a thirdparty
+	 *
+	 * @param int		$id					Id of thirdparty
+	 * @param int       $representative_id	Id of representative
+	 * @return Object|void
+	 *
+	 * @url POST {id}/representative/{representative_id}
+	 */
+	public function addRepresentative($id, $representative_id)
+	{
+		if (!DolibarrApiAccess::$user->hasRight('societe', 'creer')) {
+			throw new RestException(401);
+		}
+		$result = $this->company->fetch($id);
+		if (!$result) {
+			throw new RestException(404, 'Thirdparty not found');
+		}
+		$usertmp = new User($this->db);
+		$result = $usertmp->fetch($representative_id);
+		if (!$result) {
+			throw new RestException(404, 'User not found');
+		}
+		if (!DolibarrApi::_checkAccessToResource('societe', $this->company->id)) {
+			throw new RestException(401, 'Access not allowed for login '.DolibarrApiAccess::$user->login);
+		}
+		$result = $this->company->add_commercial(DolibarrApiAccess::$user, $representative_id);
+
+		return $result;
+	}
+
+	/**
+	 * Delete a customer representative to a thirdparty
+	 *
+	 * @param int		$id					Id of thirdparty
+	 * @param int       $representative_id	Id of representative
+	 * @return Object|void
+	 *
+	 * @url DELETE {id}/representative/{representative_id}
+	 */
+	public function deleteRepresentative($id, $representative_id)
+	{
+		if (!DolibarrApiAccess::$user->hasRight('societe', 'supprimer')) {
+			throw new RestException(401);
```

Comment:
I think we should use 403 for permission pb.
401 for bad password, 403 for no permission
https://www.permit.io/blog/401-vs-403-error-whats-the-difference

Reply:
![image](https://github.com/Dolibarr/dolibarr/assets/3624836/acafa316-6fd5-431d-9db2-50605cb34f54)

Reply:
Yes, we have to fix this to switch old code from 401 to 403.